crates/evm/core/src/utils.rs

```diff
@@ -142,7 +142,7 @@ pub fn create2_handler_register<DB: revm::Database, I: InspectorExt<DB>>(
                 .push((ctx.evm.journaled_state.depth(), call_inputs.clone()));
 
             // Sanity check that CREATE2 deployer exists.
-            let code_hash = ctx.evm.load_account(DEFAULT_CREATE2_DEPLOYER)?.0.info.code_hash;
+            let code_hash = ctx.evm.load_account(call_inputs.target_address)?.0.info.code_hash;
```

Comment:
This is changing the semantics of this code, we should use `if zk { .. } else { .. }`

Reply:
I do not have access to the inspector use_zk data from here,  I could check which deployer we are using and keep the logic the same as it was for the case the deployer is the EVM one.

Reply:
I actually think that the semantic here is a subtle bug, in particular, the `call_inputs` we see here are generated slightly above, and normally have `DEFAULT_CREATE2_DEPLOYER`, hence why it's hardcoded here too, but since these `call_inputs` are local I thought this was actually equivalent and yet allow us to use the zk value if we change `call_inputs` in the inspector (which we do)
The only failure I can see here is if an inspector edits the `call_inputs` in an erroneous way